### ISSUE TITLE
Captcha during OAuth flow causes missing_redirect_uri

### ISSUE THREAD

**Issue by mikemonaco** [FusionAuth/fusionauth-issues#2120]
### Description 
Our mobile apps use OAuth to authenticate with FusionAuth. Sometimes during the authentication flow, the user is presented with a captcha. This captcha seems to be in the style of the AWS WAF captcha. In the case where the user completes the captcha, they receive a FusionAuth error page saying:

```
Error
We're sorry, your request was malformed or was unable to
be completed for some reason. Try hitting the back button
and restarting the process to see if it fixes the probem
If you want the nerd explanation, review the following JSON body.

"error" : invalid request
"error_description" : "The request is missin..."
"error_reason": "missing_redirect_uri"
```

### Affects versions
FusionAuth version 1.38.1

### Steps to reproduce
1. Use a mobile application that has a FusionAuth OAath flow for logging in.
2. Complete the Captcha when prompted (you may have to try the flow many times in a row to get the captcha to show)
3. After the captcha is completed, you will receive an error "missing_redirect_uri" instead of redirecting back to the mobile application

### Expected behavior
The user should return to the mobile application, rather than receiving the "missing_redirect_uri" message. If the user does not get the captcha, the login and redirect work fine.

### Platform
All reports have been on iOS, but we have not been able to test Android yet.

### Community guidelines
All issues filed in this repository must abide by the [FusionAuth community guidelines](https://fusionauth.io/community/forum/topic/1000/code-of-conduct).

### Additional context
Our redirect URI is in the form of a url scheme to redirect back to a mobile application, ie. `com.company.application.auth://redirect`

**Comment by mikemonaco** [comment]
Id like to add two things. 

1) The users are logging in with a SAMLv2 IDP from the FusionAuth hosted login page.

2) We have debug logs enabled from the Application and the SAMLv2 iDP, but I do not see these errors showing in FusionAuth. They are only presented to the user. If there was a way to see more details around this error (for example: the whole request) than we might be able to debug further.

**Comment by mikemonaco** [comment]
Trying to dig using Proxyman. It seems like when the WAF imposes a captcha on the fusionauth `/samlv2/acs` endpoint we experience this issue.

**Comment by mikemonaco** [comment]
It may be loosing the `RelayState` once the captcha completes

**Comment by mikemonaco** [comment]
Update: when the POST to the samlv2/acs endpoint is caught by the AWS WAF Capatcha, and the user completes the captcha, the subsequent request to the samlv2/acs endpoint is missing the request body (namely the RelayState). Prior to the captcha, the samlv2/acs endoint had a RelayState.

Per [AWS Best Practices](https://docs.aws.amazon.com/waf/latest/developerguide/waf-captcha-and-challenge-best-practices.html):

`Use the Challenge and CAPTCHA actions only on GET text/html requests. Don't use them for POST or other request types, because browser behavior for other requests can vary and might not be able to handle the interstitials properly.`

Since the samlv2/acs endpoint is a POST, it should not be part of the Captcha ruleset that FusionAuth Cloud has implemented.

Suggestion: Reconfigure AWS WAF to

**Comment by mikemonaco** [comment]
Renamed this issue because it's a FusionAuth Cloud issue more than its a FusionAuth software issue.

**Comment by mikemonaco** [comment]
Not sure if I need to tag you but just in case: @robotdan @mooreds

**Comment by robotdan** [comment]
If you are seeing a CAPTCHA from AWS in FusionAuth Cloud it is due to rate limiting. The rate limiting threshold is rather generous in most cases and is really only intended to slow down malicious actors. 

If end users are seeing this CAPTCHA under normal use, this would not be expected. Are you proxying one to many mobile users through a single IP address?

> Suggestion: Reconfigure FusionAuth Cloud AWS WAF to only apply Captcha and Challenges to GET text/html requests.

This is a good suggestion - but I'm not sure what the alternative would be - meaning we still want to rate limit the call or block it once it has reached a pre-configured threshold. We can review the AWS documentation further to see if there is a way we can rate limit and block appropriate traffic w/out breaking the request once the user completes a CAPTCHA.

**Comment by mikemonaco** [comment]
@robotdan Yes, our use case can have many different people logging in using the same IP address (ie, they are on the same WiFi network)
